lib/github-pages-health-check/domain.rb

```diff
@@ -0,0 +1,202 @@
+module GitHubPages
+  module HealthCheck
+    class Domain < GitHubPages::HealthCheck::Checkable
+
+      attr_reader :host
+
+      def initialize(host)
+        @host = host_from_uri(host)
```

Comment:
It might be good to defensively check the input here:

```
raise ArgumentError, "Expected string, got #{host.class}" unless host.is_a? String
```

I see a lot of `undefined method for NilClass:nil` in our future without it :)